openapi-codegen/sbt-plugin/src/sbt-test/sbt-openapi-codegen/oneOf-json-roundtrip_jsoniter/src/test/scala/BinaryEndpoints.scala

```diff
@@ -0,0 +1,150 @@
+import org.apache.pekko.util.ByteString
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+import sttp.capabilities.pekko.PekkoStreams
+import sttp.client3.testing.SttpBackendStub
+import sttp.client3.{UriContext, asStringAlways}
+import sttp.monad.FutureMonad
+import sttp.tapir.generated.TapirGeneratedEndpoints
+import sttp.tapir.generated.TapirGeneratedEndpoints.{
+  PostCustomContentNegotiationBody0In,
+  PostCustomContentNegotiationBody1In,
+  PostCustomContentNegotiationBodyErrFull,
+  PostCustomContentNegotiationBodyOut,
+  PostCustomContentNegotiationBodyOutFull
+}
+import sttp.tapir.server.stub.TapirStubInterpreter
+
+import scala.collection.immutable
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, Future}
+import scala.util.Random
+
+class BinaryEndpoints extends AnyFreeSpec with Matchers {
+  private implicit val system: ActorSystem = ActorSystem()
+  private implicit val materializer: Materializer = Materializer.matFromSystem(system)
+  "binary endpoint negotiation works" in {
+    case class State(s: String)
+    var received: State = State("")
+    def iterator: Iterator[ByteString] = new Iterator[ByteString] {
+      private var linesToGo: Int = 100
+      def hasNext: Boolean = linesToGo > 0
+      def next(): ByteString = {
+        val nxt = Random.alphanumeric.take(140).mkString
+        linesToGo -= 1
+        ByteString.fromString(nxt, "utf-8")
+      }
+    }
+    def handleCsv(stream: ByteString): Unit = received = State(stream.utf8String.take(100).mkString)
+    def handleSpreadsheet(stream: ByteString): Unit = received = State(stream.utf8String.take(10).mkString)
+    def produceCsv(): ByteString = ByteString("CSV")
+    def produceSpreadsheet(): ByteString = ByteString("Spreadsheet")
+    val route1 =
+      TapirGeneratedEndpoints.postCustomContentNegotiation
+        .serverLogicSuccess({ body =>
+          val partial = body match {
+            case PostCustomContentNegotiationBody0In(csv)         => csv.map(handleCsv)
+            case PostCustomContentNegotiationBody1In(spreadsheet) => spreadsheet.map(handleSpreadsheet)
+          }
+          val running = partial.run()
+          def out: PostCustomContentNegotiationBodyOut =
+            PostCustomContentNegotiationBodyOutFull(
+              `text/csv` = () => {
+                org.apache.pekko.stream.scaladsl
+                  .Source[ByteString](immutable.Iterable.fill(100)(produceCsv()))
+                  .concat(Source.future(running).map(_ => ByteString("FIN")))
+              },
+              `application/vnd.openxmlformats-officedocument.spreadsheetml.sheet` = () => {
+                org.apache.pekko.stream.scaladsl
+                  .Source[ByteString](immutable.Iterable.fill(100)(produceSpreadsheet()))
+                  .concat(Source.future(running).map(_ => ByteString("FIN")))
+              }
+            )
```

Comment:
Since the generated code for the edge-cases can look pretty gnarly now, I think it's useful to look at the kind of binding it permits... I don't think we can do better for the user than to present this type-safe thunk-holder, at least as far as abstracting away content-type negotiation goes.

Reply:
I guess the other option for these sorts of things is to permit user injection of a codec. I think that's got a fair few issues, but can be made to work. I think it makes sense to drive that via extensions somehow... But this seemed a less intimidating prospect than failing imports on non-existent things